src/trixi_include.jl

```diff
@@ -45,13 +49,61 @@ function trixi_include(mod::Module, elixir::AbstractString; kwargs...)
     if !mpi_isparallel(Val{:MPIExt}())
         @info "You just called `trixi_include`. Julia may now compile the code, please be patient."
     end
-    Base.include(ex -> replace_assignments(insert_maxiters(ex); kwargs...), mod, elixir)
+    Base.include(ex -> replace_assignments(insert_maxiters(mapexpr(ex)); kwargs...),
+                 mod, elixir)
+end
+
+function trixi_include(mod::Module, elixir::AbstractString; kwargs...)
+    trixi_include(identity, mod, elixir; kwargs...)
 end
 
 function trixi_include(elixir::AbstractString; kwargs...)
     trixi_include(Main, elixir; kwargs...)
 end
 
+"""
+    trixi_include_changeprecision(T, [mod::Module=Main,] elixir::AbstractString; kwargs...)
+
+`include` the elixir `elixir` and evaluate its content in the global scope of module `mod`.
+You can override specific assignments in `elixir` by supplying keyword arguments,
+similar to [`trixi_include`](@ref).
+
+The only difference to [`trixi_include`](@ref) is that the precision of floating-point
+numbers in the included elixir is changed to `T`.
+More precisely, the package [ChangePrecision.jl](https://github.com/JuliaMath/ChangePrecision.jl)
+is used to convert all `Float64` literals, operations like `/` that produce `Float64` results,
+and functions like `ones` that return `Float64` arrays by default, to the desired type `T`.
+See the documentation of ChangePrecision.jl for more details.
+
+The purpose of this function is to conveniently run a full simulation with `Float32`,
+which is orders of magnitude faster on most GPUs than `Float64`, by just including
+the elixir with `trixi_include_changeprecision(Float32, elixir)`.
+Most code in the Trixi framework is written in a way that changing all floating-point
+numbers in the elixir to `Float32` manually will run the full simulation with single precision.
```

Comment:
I'm not susosure about this. For example, in Trixi.jl, we need to specify additional stuff like `RealT = Float32` etc. at several places, e.g.,

https://github.com/trixi-framework/Trixi.jl/blob/c69fe96da676f2521889fc79f68f8e6a6b5a4e7e/examples/unstructured_2d_dgsem/elixir_shallowwater_ec_float32.jl

https://github.com/trixi-framework/Trixi.jl/blob/c69fe96da676f2521889fc79f68f8e6a6b5a4e7e/examples/unstructured_2d_dgsem/elixir_shallowwater_ec_float32.jl#L33

https://github.com/trixi-framework/Trixi.jl/blob/c69fe96da676f2521889fc79f68f8e6a6b5a4e7e/examples/p4est_2d_dgsem/elixir_euler_shockcapturing_ec_float32.jl#L14

Reply:
So how do I formulate the purpose of the macro then? In TrixiParticles, we only need to change everything to `Float32` and the full simulation runs with single precision.

Maybe something like
```suggestion
Many constructors in the Trixi.jl framework are written in a way that changing all floating-point
arguments to `Float32` will change the element type to `Float32` as well.
In TrixiParticles.jl, including an elixir with this macro should be sufficient
to run the full simulation with single precision.
```

Reply:
👍